coveo-functools/coveo_functools/exceptions.py

```diff
@@ -8,3 +11,13 @@ class Flexception(CoveoFunctoolsException):  # sorry for the pun :socanadian:
 
 class UnsupportedAnnotation(Flexception, NotImplementedError):
     """When an annotation isn't supported."""
+
+
+class PayloadMismatch(Flexception):
+    """When the payload doesn't match the target type."""
+
+    def __init__(self, value: Any, hint: Any, contains: Any) -> None:
+        super().__init__(
+            f"I don't know how to fit {type(value)} "
+            f"into {hint}{' of ' + str(contains) if contains else ''}"

```

Comment:
I'm not sure if it's a good idea but you might want to add the value to the message or add the value as a separate field. When someone sees this error, they'll need to dig into the code and start debugging to figure out what the problematic value was. If it's in the message then it's easy to tell what's going on.

I'm not sure if this is a good idea because these values may contain sensitive information or they may be gigantic.

I think that if you add the value as a field then platforms like Sentry can interface it in their UI (I think). They have a layer that removes secrets which should work on that.

Reply:
Good call about the log fields, I'll add that. I had the exact same concern about sensitive or gigantic information.